core/templates/dev/head/domain/state_card/StateCardObjectFactory.js

```diff
@@ -0,0 +1,108 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating new frontend instances of State
+ * card domain objects used in the exploration player.
+ */
+
+oppia.factory('StateCardObjectFactory', [function() {
+  var StateCard = function(
+      stateName, currentParams, contentHtml, interactionHtml,
+      leadsToConceptCard, destStateName, inputResponsePairs) {
+    this._stateName = stateName;
+    this._currentParams = currentParams;
+    this._contentHtml = contentHtml;
+    this._interactionHtml = interactionHtml;
+    this._leadsToConceptCard = leadsToConceptCard;
+    this._destStateName = destStateName;
+    this._inputResponsePairs = inputResponsePairs;
+  };
+
+  StateCard.prototype.getDestStateName = function() {
+    return this._destStateName;
+  };
+
+  StateCard.prototype.getStateName = function() {
+    return this._stateName;
+  };
+
+  StateCard.prototype.getCurrentParams = function() {
+    return this._currentParams;
+  };
+
+  StateCard.prototype.getContentHtml = function() {
+    return this._contentHtml;
+  };
+
+  StateCard.prototype.getInteractionHtml = function() {
+    return this._interactionHtml;
+  };
+
+  StateCard.prototype.getOppiaResponse = function(index) {
+    return this._inputResponsePairs[index].oppiaResponse;
+  };
+
+  StateCard.prototype.getLeadsToConceptCard = function() {
+    return this._leadsToConceptCard;
+  };
+
+  StateCard.prototype.getInputResponsePairs = function() {
+    return this._inputResponsePairs;
+  };
+
+  StateCard.prototype.getLastInputResponsePair = function() {
+    return this._inputResponsePairs[this._inputResponsePairs.length - 1];
+  };
+
+  StateCard.prototype.getLastOppiaResponse = function() {
+    return (
+      this._inputResponsePairs[
```

Comment:
Also guard against getLastInputResponsePair() returning null. (Seen a production error hitting this case before.)

Reply:
Done and added tests.